Add filter logic to database

Davvos11 opened the pull request.

Davvos11 committed:
add filter logic

Davvos11 committed:
Merge branch 'main' into database

Davvos11 committed:
add day of week to filters

Davvos11 committed:
add stop name and line "number" to filters

Davvos11 committed:
add DEPARTURE filter

Davvos11 committed:
add start and end station filtering

Davvos11 committed:
Merge branch 'main' into database

Davvos11 committed:
add logic to generate ids for buslines and getting all buslines

Davvos11 committed:
add logic to generate ids for buslines and getting all buslines

VoidLeech commented on app/Database.hs:
```diff
@@ -1,35 +1,116 @@
+    show (TimeOfDay h m) = printf "%02d" h ++ ":" ++ printf "%02d" m ++ ":00"
 
-data Frequency = Frequency { value :: Int, frequency :: Int} deriving Show
+data DayOfWeek = Mon | Tue | Wed | Thu | Fri | Sat | Sun
```
I used `Data.Time.Calendar`'s `DayOfWeek` in the message format. It already has a Show instance, and you can use `weekToNumber = show . fromEnum`

Davvos11 committed:
add id's for stops and lines to the database

Davvos11 committed:
help

Davvos11 committed:
use id's in filter logic

Davvos11 committed:
add docker config

Davvos11 committed:
fix duplicates in get_stops

Davvos11 committed:
use actual data for stops and busses requests

Davvos11 committed:
WIP: attempt to implement delay requests

departure_delay is not working right now, and the attempt I made is honestly disgusting.

I have some idea's on how to get both departure and destination delay propperly (i.e. getting them in one query with a join instead of the current messy "esists" implementation).

But for now this at least shows some results.

Davvos11 committed:
Merge branch 'main' into database

Davvos11 committed:
cleanup imports
